src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ASN1.cs

```diff
@@ -61,6 +61,22 @@ internal static partial class Crypto
         [DllImport(Libraries.CryptoNative)]
         internal static extern void Asn1StringFree(IntPtr o);
 
+        internal static string GetOidValue(SafeSharedAsn1ObjectHandle asn1Object)
+        {
+            Debug.Assert(asn1Object != null);
+
+            bool added = false;
+            asn1Object.DangerousAddRef(ref added);
```

Comment:
Technically the pattern is to do this inside the try block and then have the finally conditionally do the release based on the value of added.  However, without thread aborts to worry about, this isn't necessary.